magento/utils/transform.ts

```diff
@@ -77,26 +88,65 @@ export const toProduct = (
 
 export const toOffer = (
   { price_info, extension_attributes, sku, currency_code }: MagentoProduct,
+  minInstallmentValue: number,
+  maxInstallments: number
 ): Offer[] => {
   if (!price_info) {
     return [];
   }
-  const { final_price } = price_info;
+
+  const { final_price, max_price, max_regular_price } = price_info;
   const { stock_item } = extension_attributes;
+  const possibleInstallmentsQtd =
+    Math.floor(final_price / minInstallmentValue) || 1;
+  const installments = Array.from(
+    {
+      length: Math.min(possibleInstallmentsQtd, maxInstallments),
+    },
+    (_v, i) => +(final_price / (i + 1)).toFixed(2)
+  );
```

Comment:
melhore esse código ta complicado de entender quando alguém for fazer manutenção fica complicado, minha sugestão abaixo, confere se a função que fiz ta de acordo com o objetivo pode ter errado algo

```
const calculateInstallments = (
  finalPrice: number,
  minInstallmentValue: number,
  maxInstallments: number
): number[] => {
  const possibleInstallmentsCount = Math.max(
    Math.floor(finalPrice / minInstallmentValue),
    1
  );
  const actualInstallmentsCount = Math.min(possibleInstallmentsCount, maxInstallments);

  returnArray.from({ length: actualInstallmentsCount }, (_, i) => 
    +(finalPrice / (i + 1)).toFixed(2)
  );

};
```